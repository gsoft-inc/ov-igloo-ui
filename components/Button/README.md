# @igloo-ui/button

Welcome to my `Button` component.

## Installation

```sh
npm i @igloo-ui/button
# or with yarn
yarn add @igloo-ui/button
```

Then to use the component in your code just import it!

```js
import Button from '@igloo-ui/button';
// and with css-modules
import '@igloo-ui/button/dist/button.css';
```
