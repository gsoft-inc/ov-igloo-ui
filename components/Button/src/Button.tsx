import * as React from 'react';
import classNames from 'classnames';

import './button.css';

export interface Props extends React.ComponentPropsWithoutRef<'button'> {
  // The content to dispaly inside the button
  children?: React.ReactNode;
  // Disabled the button, the user cannot click on them
  disabled?: boolean;
  //Â Sets the button in a active state
  active?: boolean;
  // Changes the size of button, giving more or less padding
  size?: 'small' | 'medium';
  // Button appearance
  appearance?: 'primary' | 'secondary' | 'premium' | 'ghost' | 'danger';
  // Add a data-test tag for automated tests
  dataTest?: string;
  // Icon to display to the left of button content
  iconLeading?: React.ReactNode;
  // Icon to display to the right of button content
  iconTrailing?: React.ReactNode;
  // Callback when clicked
  onClick?: () => void;
  // Optional prop to specify the type of the Button
  type?: 'button' | 'reset' | 'submit';
  // Add a data-intercom-target with unique id to link a components to a Product Tour step.
  intercomTarget?: string;
}

const Button: React.FunctionComponent<Props> = (props: Props) => {
  const {
    children,
    disabled = false,
    active = false,
    size = 'medium',
    appearance = 'primary',
    type = 'button',
    onClick,
    dataTest,
    iconLeading,
    iconTrailing,
    intercomTarget,
    ...rest
  } = props;

  const hasIconLeading = iconLeading !== undefined;
  const hasIconTrailing = iconTrailing !== undefined;
  const hasIcon = hasIconLeading || hasIconTrailing;

  const classes = classNames(
    'ids-btn',
    size === 'small' && 'ids-btn--small',
    active && 'ids-btn--active',
    appearance !== 'primary' && `ids-btn--${appearance}`,
    hasIcon && 'has-icon',
    hasIconLeading && 'has-icon--leading',
    hasIconTrailing && 'has-icon--trailing'
  );

  return (
    <button
      disabled={disabled}
      className={classes}
      data-test={dataTest}
      data-intercom-target={intercomTarget}
      type={type}
      onClick={onClick}
      {...rest}
    >
      {hasIconLeading && iconLeading}
      {children}
      {hasIconTrailing && iconTrailing}
    </button>
  );
};

export default Button;
