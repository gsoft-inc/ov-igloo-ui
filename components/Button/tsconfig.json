{
  "extends": "../../tsconfig.build.json",
  "include": ["src/**/*", "../../typings/**/*", "rollup.config.js"],
  "exclude": ["dist"],
  "compilerOptions": {
    "outDir": "./dist",
    "rootDir": "./src",
    "declarationDir": "./dist",
    "composite": true,
    "baseUrl": "."
  }
}
