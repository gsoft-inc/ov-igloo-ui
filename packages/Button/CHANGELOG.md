# Change Log

## 0.8.3

### Patch Changes

- 97edb0e: The button is now square when there's only an icon showing on mobile.

## 0.8.2

### Patch Changes

- 6ea531f: Updated dependency versions for react-aria packages, luxon and hopper

## 0.8.1

### Patch Changes

- f30fe4b: Color adjustments for ghost buttons

## 0.8.0

### Minor Changes

- 4846c59: Wrapped Igloo CSS in @layer to isolate it.

## 0.7.7

### Patch Changes

- cfb5619: Fixed button padding. Also removed styles on loader buttons so they don't show up on focus.
- cfb5619: Updated hopper token versions and button versions.

## 0.7.6

### Patch Changes

- 5c12940: Fixed button padding. Also removed styles on loader buttons so they don't show up on focus.

## 0.7.5

### Patch Changes

- 37260f9: Updated Button to use :focus-visible and not :focus. The old method isn't needed anymore since we no longer support IE. Also added missing line-height a ensured box-sixing is applied to all children of button.

## 0.7.4

### Patch Changes

- 7d36815: Updated hopper token versions and igloo icon versions.

## 0.7.3

### Patch Changes

- a2df55d: Updated to latest Hopper Token dependency

## 0.7.2

### Patch Changes

- 592afb0: change the border value to render on base 10 and 16.

## 0.7.1

### Patch Changes

- 9739155: Gave ColorPicker the new Wokleap look. Color now uses css variables to update its color. Fixed spacing issues in tooltip. Updated hopper token version.

## 0.7.0

### Minor Changes

- 8eeac98: Rebrand of dialog and modal / media query fix in rems

## 0.6.0

### Minor Changes

- 5402bcc: Created a "brand" toggle in the Toolbar and applied the new Workleap brand to some components. Also deprecated the Tag's appearance values and added new ones.

## 0.5.3

### Patch Changes

- 7662390: Removed the :active styles for Button since they were not needed or wanted. .ids-btn--active remains the same.

## 0.5.2

### Patch Changes

- bf4ddef: Updated storybook version and linting. Made linting fixes in these components.

## 0.5.1

### Patch Changes

- c3efaf5: Fixed tests and snapshots since react-aria generated ids changed and viewBox was added to icons.

## 0.5.0

### Minor Changes

- 357133d: Add new props for reset the size to the default on mobile

## 0.4.2

### Patch Changes

- 949db42: Added a display name to the Button so that it displays properly in storybook's code snippet.

## 0.4.1

### Patch Changes

- a41e1d4: Fix: moved destructuring of props to function parameters so that storybook will automatically show the default value in the props table. It was done in Modal because it needed to send props to useDialog. In this case, @default annotation was used in comments.

## 0.4.0

### Minor Changes

- d2e08a0: Feat: Added the ghost secondary appearance of the button.

### Patch Changes

- d5c91f5: link shared components with aliases

## 0.3.7

### Patch Changes

- 6ee9c45: Updated anything causing errors or warnings in the browser console

## 0.3.6

### Patch Changes

- 0640409: Update @igloo-ui/icons to 1.6.0
- 0640409: udpate classnames to 2.3.2

## 0.3.5

### Patch Changes

- d971143: All tests were updated to use testing library and not enzyme

## 0.3.4

### Patch Changes

- 4bd97f4: fix the hover state rendering of the disabled secondary button

## 0.3.3

### Patch Changes

- 8bc7f2c: add not-allowed cursor on disabled state

## 0.3.2

### Patch Changes

- f1611fd: Temporary quick fix on ghost danger button by adding `.ids-btn` prefix

## 0.3.1

### Patch Changes

- 4f79069: enforce button ghost danger ui

## 0.3.0

### Minor Changes

- 4de58ec: Add new appearance Ghost Danger in `Button`

### Patch Changes

- 38a4994: Update dependency on @igloo-ui/icons and other @igloo libs

## 0.2.3

### Patch Changes

- c89877e: Fix render of ghost button in ie11

## 0.2.2

### Patch Changes

- 0a86f13: Update content of README

## 0.2.1

### Patch Changes

- 3eb651d: fix the width of the button as link

## 0.2.0

### Minor Changes

- f019f88: Add the ability to display a link by an `a` tag

## 0.1.12

### Patch Changes

- ac5d81a: add forwardRef

## 0.1.11

### Patch Changes

- 8c8686e: Change @import of font.css file for a @use in SASS files

## 0.1.10

### Patch Changes

- 6d91ab6: Fix secondary button empty background on hover state

## 0.1.9

### Patch Changes

- 4a256e3: IE11 compatibility

## 0.1.8

### Patch Changes

- 1e1c6e5: Chore Update packages

## 0.1.7

### Patch Changes

- ae0a0a5: Fixed the case where the content of the Button is not text, but that we tried to display the object in the title property.

## 0.1.6

### Patch Changes

- 0df8530: Rename Props to ButtonProps

## 0.1.5

### Patch Changes

- 8450a04: Fix resizing effect when loading is complete

## 0.1.4

### Patch Changes

- 2da8010: Fix render of elements with the css var --grey-{...}

## 0.1.3

### Patch Changes

- 31593ad: Change Premium button color and focus, how buttons handle overflow, how active button handle hover and some tweaks to the dev tooling on Windows.

## 0.1.2

### Patch Changes

- c290c9a: remove `src` folder from the package

## 0.1.1

### Patch Changes

- 4ed4836: Add prepublish-ci script for build lib

## 0.1.0

### Minor Changes

- 76a63cb: - **button:** add loading effect ([2a8d84f](https://github.com/gsoft-inc/ov-igloo-ui/commit/2a8d84fa8739f170e0189d053ec4f5136ad3c4d4))
  - **button:** add of the responsive view ([364a926](https://github.com/gsoft-inc/ov-igloo-ui/commit/364a92611aae51ab555feaa6e503933ca80f1433))
  - **button:** classname for OV standard compoliance ([8ed7afb](https://github.com/gsoft-inc/ov-igloo-ui/commit/8ed7afb047f552f52fd151ef2da04be49f9908f1))
  - **button:** change the css variables name ([68f710d](https://github.com/gsoft-inc/ov-igloo-ui/commit/68f710d16e653681ff90e9124ebbafd759dfe057))
  - add declaration for `renderContent`  ([5827120](https://github.com/gsoft-inc/ov-igloo-ui/commit/58271200d14a3372dc81fd1d9189e5dfade87143))
  - add focus state ([31910e4](https://github.com/gsoft-inc/ov-igloo-ui/commit/31910e48ff15d197323666bae376116d6fe97c56))
  - remove focus when the button is clicked ([caa6d45](https://github.com/gsoft-inc/ov-igloo-ui/commit/caa6d4544979d605794b23b64fdd838af3c61a23))
  - update stylelint for OV standard compliance ([184d006](https://github.com/gsoft-inc/ov-igloo-ui/commit/184d006ecb10d8b0a19a2303f7399b54b895e992))
  - configure stylelint ([d27df67](https://github.com/gsoft-inc/ov-igloo-ui/commit/d27df672a99145eb3a134cab4549b0e0998a6d0f))

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 0.1.0-alpha.0 (2021-10-29)

### Bug Fixes

- add declaration for `renderContent`  ([5827120](https://github.com/gsoft-inc/ov-igloo-ui/commit/58271200d14a3372dc81fd1d9189e5dfade87143))
- **button:** classname for OV standard compoliance ([8ed7afb](https://github.com/gsoft-inc/ov-igloo-ui/commit/8ed7afb047f552f52fd151ef2da04be49f9908f1))
- remove focus when the button is clicked ([caa6d45](https://github.com/gsoft-inc/ov-igloo-ui/commit/caa6d4544979d605794b23b64fdd838af3c61a23))
- configure stylelint ([d27df67](https://github.com/gsoft-inc/ov-igloo-ui/commit/d27df672a99145eb3a134cab4549b0e0998a6d0f))
- update stylelint for OV standard compliance ([184d006](https://github.com/gsoft-inc/ov-igloo-ui/commit/184d006ecb10d8b0a19a2303f7399b54b895e992))

### Features

- add focus state ([31910e4](https://github.com/gsoft-inc/ov-igloo-ui/commit/31910e48ff15d197323666bae376116d6fe97c56))
- **button:** add loading effect ([2a8d84f](https://github.com/gsoft-inc/ov-igloo-ui/commit/2a8d84fa8739f170e0189d053ec4f5136ad3c4d4))
- **button:** add of the responsive view ([364a926](https://github.com/gsoft-inc/ov-igloo-ui/commit/364a92611aae51ab555feaa6e503933ca80f1433))
- **button:** change the css variables name ([68f710d](https://github.com/gsoft-inc/ov-igloo-ui/commit/68f710d16e653681ff90e9124ebbafd759dfe057))
