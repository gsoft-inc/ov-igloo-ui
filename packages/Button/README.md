# Button

Buttons are used primarily for actions.

<Example />

<ReferenceLinks />

## Installation

Install the component from your command line.

```bash
npm install @igloo-ui/button
# or with yarn
yarn add @igloo-ui/button
```

## Usage

Then to use the component in your code just import it!

```jsx
import { Button } from '@igloo-ui/button';

<Button>Primary</Button>;
```
