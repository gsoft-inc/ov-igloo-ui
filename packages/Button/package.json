{
  "name": "@igloo-ui/button",
  "version": "0.2.3",
  "main": "dist/Button.js",
  "module": "dist/Button.js",
  "types": "dist/Button.d.ts",
  "styles": "dist/button.css",
  "publishConfig": {
    "access": "public"
  },
  "repository": "git+https://github.com/gsoft-inc/ov-igloo-ui.git",
  "author": "Officevibe Inc.",
  "contributors": [
    "Franck franck.gaudin@gsoft.com"
  ],
  "license": "Apache-2.0",
  "peerDependencies": {
    "react": ">= 16.8.6"
  },
  "files": [
    "dist"
  ],
  "dependencies": {
    "@igloo-ui/tokens": "^1.1.0",
    "classnames": "^2.3.1"
  },
  "devDependencies": {
    "@igloo-ui/icons": "^0.6.2"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "IE >= 11"
  ]
}
