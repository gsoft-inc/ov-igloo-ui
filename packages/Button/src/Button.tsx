import * as React from 'react';
import cx from 'classnames';

import './button.scss';

export type Appearance =
  | 'primary'
  | 'secondary'
  | 'premium'
  | 'ghost'
  | 'danger'
  | { type: 'ghost'; variant?: 'secondary' }
  | { type: 'ghost'; variant?: 'danger' };
export type Size = 'small' | 'medium';

type AsProp<C extends React.ElementType> = {
  as?: C;
};

type PropsToOmit<C extends React.ElementType, P> = keyof (AsProp<C> & P);

type PolymorphicComponentProp<
  C extends React.ElementType,
  Props = {}
> = React.PropsWithChildren<Props & AsProp<C>> &
  Omit<React.ComponentPropsWithoutRef<C>, PropsToOmit<C, Props>>;

type PolymorphicComponentPropWithRef<
  C extends React.ElementType,
  Props = {}
> = PolymorphicComponentProp<C, Props> & { ref?: PolymorphicRef<C> };

type PolymorphicRef<C extends React.ElementType> =
  React.ComponentPropsWithRef<C>['ref'];

export interface ButtonOwnProps {
  /** The content to display inside the button */
  children?: React.ReactNode;
  /** Disabled the button, the user cannot click on them */
  disabled?: boolean;
  /** Sets the button in an active state */
  active?: boolean;
  /** Replaces button text with a spinner while a
   * background action is being performed */
  loading?: boolean;
  /** Changes the size of button, giving more or less padding */
  size?: Size;
  /** Button appearance */
  appearance?: Appearance;
  /** Add a data-test tag for automated tests */
  dataTest?: string;
  /** Icon to display to the left of button content */
  iconLeading?: React.ReactNode;
  /** Icon to display to the right of button content */
  iconTrailing?: React.ReactNode;
  /** Display only the icon in mobile */
  showOnlyIconOnMobile?: boolean;
  /** Callback when clicked */
  onClick?: () => void;
  /** Optional prop to specify the type of the Button */
  type?: 'button' | 'reset' | 'submit';
  /** Add a data-intercom-target with unique id to link a
   * component to a Product Tour step */
  intercomTarget?: string;
  /** Add a specific class to the button */
  className?: string;
  //** Return to the default size on mobile */
  resetSizeOnMobile?: boolean;
}

type ButtonProps<C extends React.ElementType> = PolymorphicComponentPropWithRef<
  C,
  ButtonOwnProps
>;

type ButtonComponent<C extends React.ElementType = 'button'> = {
  <T extends React.ElementType = C>(
    props: ButtonProps<T>
  ): React.ReactElement | null;

  displayName?: string;
};

const Button: ButtonComponent = React.forwardRef(
  <C extends React.ElementType = 'button'>(
    {
      children,
      disabled = false,
      active = false,
      loading = false,
      size = 'medium',
      appearance = 'primary',
      type = 'button',
      onClick,
      dataTest,
      iconLeading,
      iconTrailing,
      showOnlyIconOnMobile,
      resetSizeOnMobile = false,
      intercomTarget,
      className,
      as,
      ...rest
    }: ButtonProps<C>,
    ref?: PolymorphicRef<C>
  ) => {
    const hasIconLeading = iconLeading !== undefined;
    const hasIconTrailing = iconTrailing !== undefined;
    const hasIcon = hasIconLeading || hasIconTrailing;
    const childrenIsAString =
      typeof children === 'string' || children instanceof String;

    const normalizeAppearanceClass = (appearance: Appearance): {} => {
      if (typeof appearance === 'object') {
        return {
          [`ids-btn--${appearance.type}`]: !appearance.variant,
          [`ids-btn--${appearance.type}-${appearance.variant}`]:
            appearance.variant,
        };
      }

      return {
        [`ids-btn--${appearance}`]: appearance !== 'primary',
      };
    };

    const appearanceClasses = normalizeAppearanceClass(appearance);

    const classes = cx('ids-btn', className, appearanceClasses, {
      'ids-btn--small': size === 'small',
      'ids-btn--active': active,
      'ids-btn--loading': loading,
      'ids-btn--disabled': as === 'a' && disabled,
      'ids-btn--mobile': showOnlyIconOnMobile,
      'has-icon': hasIcon,
      'has-icon--leading': hasIconLeading,
      'has-icon--trailing': hasIconTrailing,
      'ids-btn--size-mobile': resetSizeOnMobile,
    });

    const renderContent = (): JSX.Element => {
      return (
        <>
          {hasIconLeading && iconLeading}
          {showOnlyIconOnMobile ? (
            <span
              className={cx('ids-btn__label', {
                'is-hidden': loading,
              })}
            >
              {children}
            </span>
          ) : (
            <span
              className={cx({
                'is-hidden': loading,
              })}
            >
              {children}
            </span>
          )}
          {hasIconTrailing && iconTrailing}
        </>
      );
    };

    const Component = as || 'button';

    return (
      <Component
        disabled={disabled}
        className={classes}
        data-test={dataTest}
        data-intercom-target={intercomTarget}
        type={type}
        title={childrenIsAString ? children?.toString() : ''}
        onClick={onClick}
        ref={ref}
        {...rest}
      >
        {loading && <div className="ids-loader" />}
        {renderContent()}
      </Component>
    );
  }
);

Button.displayName = 'Button'; // This fixed issue OV-42946

export default Button;
