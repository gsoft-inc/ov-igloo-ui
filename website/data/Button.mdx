# Button
Buttons are used primarily for actions.

<div>
  <Button>Primary</Button>
  <Button appearance="secondary">Secondary</Button>
  <Button appearance="premium">Premium</Button>
  <Button appearance="danger">Danger</Button>
  <Button appearance="ghost">Ghost</Button>
</div>

## Installation
Install the component from your command line.
```bash
npm install @igloo-ui/button
# or with yarn
yarn add @igloo-ui/button
```

## Usage
Import all parts and piece them together.

```jsx
import { Button } from '@igloo-ui/button';

<Button>Primary</Button>
```

## API
| Prop | Type | Default | Description |
|:---|:---|:---|:---|
| appearance | string | - | - |


