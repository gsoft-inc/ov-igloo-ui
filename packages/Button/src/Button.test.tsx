/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
import Plus from '@igloo-ui/icons/dist/Plus';

import Button, { ButtonOwnProps } from './Button';

const setUp = (props: ButtonOwnProps = {}) => {
  return render(
    <Button dataTest="ids-btn" {...props}>
      Click me
    </Button>
  );
};

describe('Button Component', (): void => {
  test('It should render without errors', (): void => {
    setUp();
    const wrapper = screen.getByRole('button');
    expect(wrapper).toBeInTheDocument();
  });

  test('It should render a button with icon', (): void => {
    const container = setUp({
      iconLeading: <Plus size="small" />,
    }).container;
    const svg = container.querySelector('svg');
    expect(svg).toBeInTheDocument();
  });

  test('It should render a primary appearance', (): void => {
    const fragment = setUp().asFragment();
    expect(fragment).toMatchSnapshot();
  });

  test('It should render a secondary appearance', (): void => {
    const fragment = setUp({ appearance: 'secondary' }).asFragment();
    expect(fragment).toMatchSnapshot();
  });

  test('It should render a premium appearance', (): void => {
    const fragment = setUp({ appearance: 'premium' }).asFragment();
    expect(fragment).toMatchSnapshot();
  });

  test('It should render a danger appearance', (): void => {
    const fragment = setUp({ appearance: 'danger' }).asFragment();
    expect(fragment).toMatchSnapshot();
  });

  test('It should render a ghost appearance', (): void => {
    const fragment = setUp({ appearance: 'ghost' }).asFragment();
    expect(fragment).toMatchSnapshot();
  });

  test('It should render a button with a tag', (): void => {
    const { container } = render(
      <Button
        as="a"
        href="://igloo.officevibe.design"
        target="_blank"
        // @ts-ignore
        name="click me"
      >
        Click me
      </Button>
    );

    const link = container.querySelector('a');
    expect(link).toBeInTheDocument();
  });
});
