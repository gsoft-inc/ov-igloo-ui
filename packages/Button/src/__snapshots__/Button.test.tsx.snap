// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component It should render a danger appearance 1`] = `
<DocumentFragment>
  <button
    class="ids-btn ids-btn--danger"
    data-test="ids-btn"
    title="Click me"
    type="button"
  >
    <span
      class=""
    >
      Click me
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button Component It should render a ghost appearance 1`] = `
<DocumentFragment>
  <button
    class="ids-btn ids-btn--ghost"
    data-test="ids-btn"
    title="Click me"
    type="button"
  >
    <span
      class=""
    >
      Click me
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button Component It should render a premium appearance 1`] = `
<DocumentFragment>
  <button
    class="ids-btn ids-btn--premium"
    data-test="ids-btn"
    title="Click me"
    type="button"
  >
    <span
      class=""
    >
      Click me
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button Component It should render a primary appearance 1`] = `
<DocumentFragment>
  <button
    class="ids-btn"
    data-test="ids-btn"
    title="Click me"
    type="button"
  >
    <span
      class=""
    >
      Click me
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button Component It should render a secondary appearance 1`] = `
<DocumentFragment>
  <button
    class="ids-btn ids-btn--secondary"
    data-test="ids-btn"
    title="Click me"
    type="button"
  >
    <span
      class=""
    >
      Click me
    </span>
  </button>
</DocumentFragment>
`;
